### PR TITLE
enable nullable on TxPool project

### PR DESCRIPTION
**Pull request by smartprogrammer93**
Part of investigation into #4722

## Changes:
- Enabled Nullable annotations in TxPool project

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [ ] Yes
- [x] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by smartprogrammer93** [comment]
Sorry had to do a new PR because of a bad branch name on my end, but i applied your comments @LukaszRozmej

**Review by LukaszRozmej — COMMENTED** [review]
Why are you updating submodule?

File: src/Nethermind/Nethermind.JsonRpc/Data/TransactionForRpc.cs [hunk 1]
```diff
@@ -129,7 +129,8 @@ public T ToTransactionWithDefaults<T>(ulong? chainId = null) where T : Transacti
             Type = Type,
             AccessList = TryGetAccessList(),
             ChainId = chainId,
-            DecodedMaxFeePerGas = MaxFeePerGas ?? 0
+            DecodedMaxFeePerGas = MaxFeePerGas ?? 0,
+            Hash = Hash ?? null
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.JsonRpc/Data/TransactionForRpc.cs:133]: `?? null` looks redundant

File: src/Nethermind/Nethermind.Core.Test/Builders/TransactionBuilder.cs [hunk 2]
```diff
@@ -51,6 +51,11 @@ public TransactionBuilder<T> WithHash(Keccak? hash)
             TestObjectInternal.Hash = hash;
             return this;
         }
+        public TransactionBuilder<T> CalculateHash()
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Core.Test/Builders/TransactionBuilder.cs:54]: Is this needed/used?

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**